lib/config/config-loader.js

```diff
@@ -173,26 +173,27 @@ async function loadConfigFile(filePath, allowTS) {
      */
     if (allowTS && isTS && !isDeno && !isBun) {
 
-        const createJiti = await import("jiti").then(jitiModule => (typeof jitiModule?.createJiti === "function" ? jitiModule.createJiti : jitiModule.default), () => {
+        // eslint-disable-next-line no-use-before-define -- `ConfigLoader.loadJiti` can be overwritten for testing
+        const { createJiti } = await ConfigLoader.loadJiti().catch(() => {
             throw new Error("The 'jiti' library is required for loading TypeScript configuration files. Make sure to install it.");
         });
 
+        // `createJiti` was added in jiti v2.
+        if (typeof createJiti !== "function") {
+            throw new Error("You are using an outdated version of the 'jiti' library. Please update to the latest version of 'jiti' to ensure compatibility and access to the latest features.");
+        }
+
         /*
          * Disabling `moduleCache` allows us to reload a
          * config file when the last modified timestamp changes.
          */
 
         const jiti = createJiti(__filename, { moduleCache: false, interopDefault: false });
-
-        if (typeof jiti?.import !== "function") {
-            throw new Error("You are using an outdated version of the 'jiti' library. Please update to the latest version of 'jiti' to ensure compatibility and access to the latest features.");
-        }
-
         const config = await jiti.import(fileURL.href);
 
         importedConfigFileModificationTime.set(filePath, mtime);
 
-        return config?.default ?? config;
+        return config.default ?? config;
```

Comment:
Without the `?`, a config like this would result in `TypeError: Cannot read properties of undefined (reading 'default')`:

```ts
// eslint.config.cts
module.exports = undefined;
```

Reply:
is that a valid config/usecase?

Reply:
Thanks! I reverted the change and added unit tests for this case in e5d3a9b3979b1a21e4f518b26b019590ff0d2b87.